File: 3708-zigzag-grid-traversal-with-skip/3708-zigzag-grid-traversal-with-skip.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* zigzagTraversal(int** grid, int gridSize, int* gridColSize, int* returnSize) {
    int area = gridSize * (*gridColSize);
    int* ans = (int*)malloc((area + 1) / 2 * sizeof(int));
    *returnSize = (area + 1) / 2;
    int pt = 0;
    for (int i = 0; i < gridSize; i++) {
        if (i % 2 == 0) {
            for (int j = 0; j < *gridColSize; j++) {
                if ((pt & 1) == 0)
                    ans[pt / 2] = grid[i][j];
            
                pt++;
            }
        } else {
            for (int j = *gridColSize - 1; j >= 0; j--) {
                if ((pt & 1) == 0) 
                    ans[pt / 2] = grid[i][j];

                pt++;
            }
        }
    }
    return ans;
}

File: 3708-zigzag-grid-traversal-with-skip/3708-zigzag-grid-traversal-with-skip.cs
public class Solution {
    public IList<int> ZigzagTraversal(int[][] grid) {
        var ans = new List<int>();
        var cnt = 0;
        var n = grid.Length; 
        for (var i = 0; i < n; i++) {
            if ((i & 1) != 0) {
                for (var j = grid[i].Length - 1; j >= 0; j--) {
                    if ((cnt & 1) == 0)
                        ans.Add(grid[i][j]);
                    cnt++;
                }
            } else {
                for (var j = 0; j < grid[i].Length; j++) {
                    if ((cnt & 1) == 0)
                        ans.Add(grid[i][j]);
                    cnt++;
                }
            }
        }
        return ans;
    }
}